e2e/scenarios/kops-simple

```diff
@@ -0,0 +1,145 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+set -x
+
+REPO_ROOT=$(git rev-parse --show-toplevel)
+cd ${REPO_ROOT}
+cd ..
+WORKSPACE=$(pwd)
+
+# Create bindir
+BINDIR=${WORKSPACE}/bin
+export PATH=${BINDIR}:${PATH}
+mkdir -p "${BINDIR}"
+
+# Build kOps
+pushd ${WORKSPACE}/kops
+GOBIN=${BINDIR} go install k8s.io/kops/cmd/kops
+GOBIN=${BINDIR} make test-e2e-install
+popd
+
+if [[ -z "${K8S_VERSION:-}" ]]; then
+  K8S_VERSION="$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)"
+fi
+
+CLOUD_PROVIDER="gce"
+echo "CLOUD_PROVIDER=${CLOUD_PROVIDER}"
+
+if [[ -z "${CLUSTER_NAME:-}" ]]; then
+  SCRIPT_NAME=$(basename $0 .sh)
+  CLUSTER_NAME="${SCRIPT_NAME}.k8s.local"
+fi
+echo "CLUSTER_NAME=${CLUSTER_NAME}"
+
+if [[ -z "${WORKDIR:-}" ]]; then
+  WORKDIR="${WORKSPACE}/clusters/${CLUSTER_NAME}"
+fi
+mkdir -p "${WORKDIR}"
+
+if [[ -z "${GCP_PROJECT}" ]]; then
+  GCP_PROJECT=$(gcloud config get project)
+fi
+echo "GCP_PROJECT=${GCP_PROJECT}"
+
+if [[ -z "${KOPS_STATE_STORE}" ]]; then
+  KOPS_STATE_STORE="gs://kops-state-${GCP_PROJECT}"
+fi
+echo "KOPS_STATE_STORE=${KOPS_STATE_STORE}"
+export KOPS_STATE_STORE
+
+if [[ -z "${IMAGE_REPO}" ]]; then
+  IMAGE_REPO="gcr.io/${GCP_PROJECT}"
+fi
+echo "IMAGE_REPO=${IMAGE_REPO}"
+
+cd ${REPO_ROOT}
+if [[ -z "${IMAGE_TAG}" ]]; then
+  IMAGE_TAG=$(git rev-parse --short HEAD)-$(date +%Y%m%dT%H%M%S)
+fi
+echo "IMAGE_TAG=${IMAGE_REPO}"
+
+# Build and push cloud-controller-manager
+cd ${REPO_ROOT}
+export KO_DOCKER_REPO="${IMAGE_REPO}"
+ko build --tags ${IMAGE_TAG} --base-import-paths --push=true ./cmd/cloud-controller-manager/
+
+# TODO: Only if running in a test project?
+gsutil iam ch allAuthenticatedUsers:objectViewer gs://artifacts.${GCP_PROJECT}.appspot.com
+
+if [[ -z "${ADMIN_ACCESS}" ]]; then
+  ADMIN_ACCESS="0.0.0.0/0" # Or use your IPv4 with /32
+fi
+echo "ADMIN_ACCESS=${ADMIN_ACCESS}"
+
+create_args="--networking calico"
+if [[ -n "${ZONES:-}" ]]; then
+    create_args="${create_args} --zones=${ZONES}"
+fi
+
+cp "${REPO_ROOT}/packages/manifests/direct/manifest.yaml" "${WORKDIR}/cloud-provider-gcp.yaml"
+sed -i -e "s@k8scloudprovidergcp/cloud-controller-manager:latest@${IMAGE_REPO}/cloud-controller-manager:${IMAGE_TAG}@g" "${WORKDIR}/cloud-provider-gcp.yaml"
+create_args="${create_args} --add=${WORKDIR}/cloud-provider-gcp.yaml"
+
+
+# Enable cluster addons, this enables us to replace the built-in manifest
+KOPS_FEATURE_FLAGS="ClusterAddons,${KOPS_FEATURE_FLAGS}"
+echo "KOPS_FEATURE_FLAGS=${KOPS_FEATURE_FLAGS}"
+
+KUBETEST2_ARGS="kops --kops-binary-path=${BINDIR}/kops"
+KUBETEST2_ARGS="${KUBETEST2_ARGS} -v=2 --cloud-provider=${CLOUD_PROVIDER}"
```

Comment:
How will this affect kubelet/kcm? For kubelet and KCM we want to set `--cloud-provider=external`.

Reply:
So yes, I realize this is a little confusing.: These are the _kops_ arguments.  I obviously need a README here, but the idea is that we can test with kOps (and hopefully other production tools), and cloud-provider-gcp should control the manifest it is verified to run with (e.g. do we need nodes patch permission, as above).

The kops `--cloud-provider` flag did originally map to kubelet and kcm, but as of recent versions of k8s we will use cloud-provider=external there.  The flag is the user saying what cloud they want, and then kOps figures out the flags for the various components.

In kOps, I'd like to get us to a more modular approach where we don't have to create our own manifests for all the components, but rather cloud-provider-gcp can specify how it wants to be run, can easily test that manifest with kOps + other tools, and then kOps (and other tools) can run with that tested manifest.

Reply:
I think a comment or README would be good. Also once we switch the disable cloud provider feature flag to beta, the KCM, KAS and Kubelet will all crash if you provide a gcp/gce cloud provider flag.

Reply:
Added a comment!  And yes, kOps will map the flags to wherever they move to over time :-)